network/transport/v2/handlers.go

```diff
@@ -248,21 +248,20 @@ func (p *protocol) handleGossip(peer transport.Peer, envelope *Envelope) error {
 	// TODO swap for trace logging
 	log.Logger().Infof("received %d new transaction references via Gossip", len(refs))
 
-	// send State if node is missing more refs than referenced in this Gossip
-	tempXor := xor.Xor(refs...)
-	if msg.LC >= clock && !tempXor.Equals(peerXor) {
-		// TODO swap for trace logging
-		log.Logger().Infof("xor is different from peer=%s and peers clock is equal or higher", peer.ID)
-		return p.sender.sendState(peer.ID, xor, clock)
-	}
-
 	// request missing refs
-	if len(refs) > 0 {
+	tempXor := xor.Xor(refs...)
```

Comment:
Maybe some extra comments for understanding?
```suggestion
	// If our DAG is just missing the TXs from the gossip to get in sync with the peer's DAG, send TransactionListQuery.
	// Test this by XORing the TX refs from the gossip message with our DAG's XOR (should then equal peer DAG's XOR).
	tempXor := xor.Xor(refs...)
```